scripts/common_test.py

```diff
@@ -451,11 +490,11 @@ def mock_exit(unused_code):
         with sleep_swap, is_port_in_use_swap, exit_swap:
             common.wait_for_port_to_be_in_use(9999)
 
-    def test_permissions_of_file(self):
+    def test_permissions_of_file(self) -> None:
         root_temp_dir = tempfile.mkdtemp()
         temp_dirpath = tempfile.mkdtemp(dir=root_temp_dir)
         temp_file = tempfile.NamedTemporaryFile(dir=temp_dirpath)
-        temp_file.name = 'temp_file'
+        setattr(temp_file, 'name', 'temp_file')
```

Comment:
Why is this needed?

Reply:
probably the reason is we were trying to write a read-only object attribute which was producing this error
### error: Property "name" defined in "IO" is read-only [misc]
and to get rid of this I used `setattr`. Please correct me if I am wrong.

Reply:
Can we rather use `StringIO` here?

Reply:
Nope I guess because here the type of `temp_file` is IO[bytes].

Reply:
But why?

Reply:
I am not getting how should I use `StringIO` here. Can you please explain a bit?

Reply:
Probably, the reason is the `name` attribute is read-only, and to set the attribute value, we need to use `setattr`. And as we have discussed here https://github.com/oppia/oppia/pull/15104, I am also adding some comments for this.

Reply:
Please update the code comment as per my suggestion in #15104

Reply:
done.